openpype/host/__init__.py

```diff
@@ -0,0 +1,13 @@
+from .host import (
+    HostImplementation,
+    IWorkfileHost,
+    ILoadHost,
+    INewPublisher,
+)
+
+__all__ = (
+    "HostImplementation",
```

Comment:
Finding it a bit odd to include "Implementation" into the class name. It's an programming concept.. and as the first line of the class comment states: 
"Host is pipeline implementation of DCC application". So it feels a bit redundant. 
Any issue with simply calling it "Host" ?

Reply:
I'm not againts but `Host` is too short (for my taste), maybe `BaseHost` or `HostBase`?

Reply:
> Any issue with simply calling it "Host" ?

I personally felt the same as @icyvapor - I'd prefer shorter if possible.
Same for `MayaHostImplementation` - I'm not sure why it couldn't be called `MayaHost`.
From a class naming convention I'd understand `MayaHost` better than `MayaHostImplementation`. The latter almost sounds like there's still another `MayaHost` class somewhere in the code base.

Reply:
Changed to `HostBase` and `MayaHost`, is it ok? Any other comments?

Reply:
The only problem with `host` as a word, that it's super hard to find it in the code base when you need to trace it's usages. `HostBase` and `MayaHost` is pretty good and specific enough not to have millions of false positives when searching through codebase

Reply:
Would `AsbtractHost` make sense too? I'm also fine with `HostBase` by the way.

Reply:
Only abstract part of the class is `name` property. In future is planned to have more pre-implemented functionality thus I'm more in `HostBase`.

Reply:
Sorry guys, soon after the comment I was put out of commission for the rest of the week by some health issue. Thanks for addressing the comment.